src/intern.ts

```diff
@@ -1,219 +1,151 @@
-          ent = entset[id]
-          for (let p in q) {
-            let qv = q[p] // query val
-            let ev = ent[p] // ent val
-
-            if (-1 === p.indexOf('$')) {
-              if (Array.isArray(qv)) {
-                if (-1 === qv.indexOf(ev)) {
-                  continue next_ent
-                }
-              } else if (intern.is_object(qv)) {
-                // mongo style constraints
-                if (
-                  (null != qv.$ne && qv.$ne == ev) ||
-                  (null != qv.$gte && qv.$gte > ev) ||
-                  (null != qv.$gt && qv.$gt >= ev) ||
-                  (null != qv.$lt && qv.$lt <= ev) ||
-                  (null != qv.$lte && qv.$lte < ev) ||
-                  (null != qv.$in && -1 === qv.$in.indexOf(ev)) ||
-                  (null != qv.$nin && -1 !== qv.$nin.indexOf(ev)) ||
-                  false
-                ) {
-                  continue next_ent
-                }
-              } else {
-                if (intern.is_date(qv)) {
-                  if (!(intern.is_date(ev) && intern.eq_dates(qv, ev))) {
-                    continue next_ent
-                  }
-                } else if (qv !== ev) {
-                  continue next_ent
-                }
-              }
-            }
-          }
-          ent = qent.make$(ent)
-          list.push(ent)
-        }
-      }
+    const args = {
+      table_name: ent_table,
+      data: data,
+      id: q
     }
 
-    // Always sort first, this is the 'expected' behaviour.
-    if (null != q && q.sort$) {
-      let sf: any
-      for (sf in q.sort$) {
-        break
-      }
+    const query = Q.upsert(args)
+    return query
+  }
 
-      let sd = q.sort$[sf] < 0 ? -1 : 1
-      list = list.sort(function (a, b) {
-        return sd * (a[sf] < b[sf] ? -1 : a[sf] === b[sf] ? 0 : 1)
-      })
-    }
+  static tablenameUtil(ent: any) {
+    const canon = ent.canon$({ object: true })
 
-    // Skip before limiting.
-    if (null != q && q.skip$ && q.skip$ > 0) {
-      list = list.slice(q.skip$)
-    }
+    return (canon.base ? canon.base + '_' : '') + canon.name
+  }
 
-    // Limited the possibly sorted and skipped list.
-    if (null != q && q.limit$ && q.limit$ >= 0) {
-      list = list.slice(0, q.limit$)
+  static async isNew(ent: any) {
+    const isNew = await intern.findKnex(ent, ent.id)
+
+    if (isNew) {
+      return true
     }
 
-    // Prune fields
-    if (null != q && q.fields$) {
-      for (let i = 0; i < list.length; i++) {
-        let entfields = list[i].fields$()
-        for (let j = 0; j < entfields.length; j++) {
-          if ('id' !== entfields[j] && -1 == q.fields$.indexOf(entfields[j])) {
-            delete list[i][entfields[j]]
-          }
-        }
+    return false
+  }
+
+  static getConfig(spec: any) {
+    let conf
+
+    if ('string' === typeof spec) {
+      const urlM = /^postgres:\/\/((.*?):(.*?)@)?(.*?)(:?(\d+))?\/(.*?)$/.exec(spec)
```

Comment:
Please see if you could use `url.parse` here instead. E.g.
https://github.com/brianc/node-postgres/tree/master/packages/pg-pool#create

Reply:
actually since, knex works with multiple databases, there should be no need for this at all
@Jose-Perigolo you should have a `dbconf` option that can be used to provide database specific config - we don't care what is inside it

Reply:
I thought we were going to use knex-store just for postgre, I can do that yes, it will be better